src/views/EditUser.js

```diff
@@ -144,38 +265,109 @@ export const EditUser = (props) => {
-                            </TableRow>)}
-                        </TableBody>
-                    </Table>}
+            <div><Link to="/users" className={classes.link_button}><Button><ArrowBackIcon /></Button></Link></div>
+            <div className={classes.user_info_container}>
+                <div style={{ display: 'flex', flexDirection: 'column' }}>
+                    <Paper className={classes.user_info}>
+                        <h2>Basic Information</h2>
+                        <div><Typography>User Name: {currentUser}</Typography></div>
+                        <div style={{ display: 'flex', alignItems: 'center'}}><Typography>User Type: </Typography>
+                            <Select style={{ marginLeft: '10px'}} value={userType.value} onChange={(e) => updateUserType(e.target.value)}>
+                                <MenuItem value="rodsuser">rodsuser</MenuItem>
+                                <MenuItem value="rodsadmin">rodsadmin</MenuItem>
+                                <MenuItem value="groupadmin">groupadmin</MenuItem>
+                            </Select>
+                        </div>
+                    </Paper>
+                    <br />
+                    <Paper className={classes.user_info}>
+                        <h2>Password Management</h2>
+                        <FormControl error={password.status === 'Password can not be empty.'} className={classes.margin}>
+                            <InputLabel htmlFor="user-new-password">Create New Password</InputLabel>
+                            <Input
+                                id="user-new-password"
+                                type={password.showPassword ? "text" : "password"}
+                                value={password.newPassword}
+                                onChange={(e) => setPassword({ ...password, newPassword: e.target.value })}
+                                endAdornment={
+                                    password.newPassword !== '' && <InputAdornment position="end">
+                                        <IconButton
+                                            aria-label="toggle password visibility"
+                                            onClick={() => setPassword({ ...password, showPassword: !password.showPassword })}
+                                        >
+                                            {password.showPassword ? <VisibilityOff /> : <Visibility />}
+                                        </IconButton>
+                                    </InputAdornment>
+                                }
+                            />
+                            <FormHelperText id="user-new-password-error">{password.status === 'Password can not be empty.' ? password.status : ''}</FormHelperText>
```

Comment:
it should be up to the server's policy as to whether the passwords CAN be empty.

let's change this to give a confirmation box if the passwords are empty.
 - dialog text.... "WARNING: Passwords are empty."
 - two buttons... "Cancel" and "Confirm"

we want to prevent accidental submission of an empty form.

Reply:
roger roger